File: go-movies-front-end/src/components/Movie.js
import React, { useEffect, useState } from "react";
import { useParams } from "react-router-dom";




const Movie = () => {
  const [movie, setMovie] = useState({});
  let { id } = useParams();

  useEffect(() => {
    let myMovie = 
      {
        id: 1,
        title: "Highlander",
        release_date: "1986-03-07",
        runtime: 116,
        mpaa_rating: "R",
        description:
          "An immortal Scottish swordsman faces off with other immortal warriors in order to obtain a coveted ability.",
      }
    setMovie(myMovie);
  }, [id])

  return (
    <div>
      <h2>Movie: {movie.title}</h2>
      <small><em>{movie.release_date}, {movie.runtime} minutes, Rated {movie.mpaa_rating}</em></small>
      <hr />
      <p>{movie.description}</p>
    </div>
  );
};

export default Movie;

File: go-movies-back-end/internal/models/Movie.go
package models

import "time"

type Movie struct {
	ID          int64     `json:"id"`
	Title       string    `json:"title"`
	Description string    `json:"description"`
	ReleaseDate time.Time `json:"release_date"`
	Runtime     int64     `json:"runtime"`
	MPAARating  string    `json:"mpaa_rating"`
	Image       string    `json:"image"`
	CreatedAt   time.Time `json:"-"`
	UpdatedAt   time.Time `json:"-"`
}
